src/tests/backend.ml

```diff
@@ -40,7 +40,7 @@ module Serialize = struct
 end
 
 module Fp (P : sig
-  val order : Bignum_bigint.t
+  val characteristic : Bignum_bigint.t
```

Comment:
I would keep order. It is not equivalent when we have an extension.

Reply:
FWIW my goal here is to just make a prime field, the use of the Bigint type as the implementation wouldn't even allow for anything else. Also want to note that `Bigint_intf` requires a `from_field` method which also doesn't make any sense for an extension. 

The fact that `Bigint_intf` relies on a field type and not the other way around is already a weird smell